File: backend/App.Domain/ParticipantEvent.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Base.Domain;

namespace App.Domain;

public class ParticipantEvent : BaseEntityIdMetadata
{
    public DateTime RegisterDateTime { get; set; }
    
    public Guid? PersonId { get; set; }
    public Person? Person { get; set; }
    
    public Guid? FirmId { get; set; }
    public Firm? Firm { get; set; }
    
    public Guid? EventId { get; set; }
    [JsonIgnore]
    public Event? Event { get; set; }
    
    [MaxLength(5000)]
    public string AdditionalNotes { get; set; } = default!;
    
    public Guid? PaymentMethodId { get; set; }
    public PaymentMethod? PaymentMethod { get; set; }
}

File: frontend/entities/ParticipantEvent.ts
import Firm from "./Firm";
import PaymentMethod from "./PaymentMethod";
import Person from "./Person";
import Event from "./Event";


export default class ParticipantEvent {
    id: string;
    registerDateTime: string;
    personId: string | null;
    person: Person | null;
    firmId: string | null;
    firm: Firm | null;
    paymentMethodId: string;
    paymentMethod: PaymentMethod | null;
    eventId: string;
    event: Event | null;

    constructor(
        id: string,
        registerDateTime: string,
        personId: string | null,
        person: Person | null,
        firmId: string | null,
        firm: Firm | null,
        paymentMethodId: string,
        paymentMethod: PaymentMethod | null,
        eventId: string,
        event: Event | null
    ) {
        this.id = id;
        this.registerDateTime = registerDateTime;
        this.personId = personId;
        this.person = person;
        this.firmId = firmId;
        this.firm = firm;
        this.paymentMethodId = paymentMethodId;
        this.paymentMethod = paymentMethod;
        this.eventId = eventId
        this.event = event
    }
}
